xtable-core/src/main/java/org/apache/xtable/iceberg/IcebergConversionSource.java

```diff
@@ -247,7 +247,7 @@ public boolean isIncrementalSyncSafeFrom(Instant instant) {
     long timeInMillis = instant.toEpochMilli();
     Table iceTable = getSourceTable();
     boolean doesInstantOfAgeExists = false;
-    Long targetSnapshotId = null;
```

Comment:
Thanks for the PR @the-other-tim-brown. This fix seems to be addressing another issue. I will let @brishi19791, who reported the original issue, to confirm. 
My understanding is that iceberg api may not return ordered snapshots as the spec does not mandate it. Also, the snapshot order seems to engine dependedent. Whereas the while loops in `isIncrementalSyncSafeFrom` is order sensitive, it may break prematurely.

Reply:
If my observation is correct, we have two options. 1) @brishi19791 creates another PR with his fix and we merge both these, or 2) merge @brishi19791's PR in this branch and merge the two fixes together. WDYT?

Reply:
@brishi19791 can you add the test updates from this branch in your PR? if so, we can close this one